### PR TITLE
[batch] Fix billing projects query to be fast with lateral joins

### PR DESCRIPTION
**Pull request by jigold**
This query is the slowest one we have right now and is making the billing projects pages to be slow. I saw substantial speed improvements using lateral joins.

- For querying all open billing projects for a given user, it went from 1.8 seconds to 0.87 seconds.
- For querying all open billing projects (i.e. billing projects page), it went from 5.5 seconds to 2.6 seconds.

The speed of the queries should go down further once the aggregated billing project user resources table has been deduped and further improvements after compaction.

### CODE REVIEW

File: batch/batch/utils.py [hunk 1]
```diff
@@ -170,6 +167,8 @@ def record_to_dict(record):
             record['users'] = json.loads(record['users'])
         return record
 
+    log.info(f'sql {sql} args {args}')
```
Comment (daniel-goldstein) [inline batch/batch/utils.py:170]: Was this for debugging?

**Review by daniel-goldstein — DISMISSED** [review]
My apologies for letting this fall off my plate. This is a great improvement, just one log message that should get deleted

**Comment by danking** [comment]
Neat! Where'd you learn about lateral joins? I had not heard about them until now.

**Review by daniel-goldstein — APPROVED** [review]

**Comment by jigold** [comment]
I got frustrated with the query speeds, tried to figure out how I'd go and implement it efficiently if I were coding this from scratch, searched google for awhile and finally came across what I needed on either Stack Overflow or the MySQL docs. It's a pretty recent feature addition. v8.0.14.

**Pull request merged**